src/main/java/org/jabref/logic/importer/fileformat/CiteSeerParser.java

```diff
@@ -0,0 +1,100 @@
+package org.jabref.logic.importer.fileformat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.CookieHandler;
+import java.net.CookieManager;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class CiteSeerParser implements Parser {
+
+//    private static final Pattern PAPERID = Pattern.compile("pid/[0-9a-zA-Z]+", Pattern.MULTILINE);
+
+    @Override
+    public List<BibEntry> parseEntries(InputStream inputStream) throws ParseException {
+        List<BibEntry> bibEntries;
+        try {
+            bibEntries = parseCiteSeerResponse(inputStream);
+        } catch (FetcherException e) {
+            throw new ParseException("Could not parse CiteSeer data, ", e);
+        }
+        return bibEntries;
+    }
+
+    private void appendData(String data, BibEntry entry, Pattern pattern, Field field) {
+        Matcher matcher = pattern.matcher(data);
+        if (matcher.find()) {
+            entry.setField(field, matcher.group(1));
+        }
+    }
+
+    private List<BibEntry> parseCiteSeerResponse(InputStream inputStream) throws FetcherException {
```

Comment:
I think, you should directly parse JSONOBject into the list. This way, you save the conversion from JSONObject to STring and back.

--> Implement `SearchBasedFetcher`  instead of `SearchBasedParserFetcher` in CiteSeer.

Your implementation does not make use at all the "magic" of the method `getBibEntries`.

Just implement `List<BibEntry> performSearch(QueryNode luceneQuery) throws FetcherException;` in `CiteSeer`.